### PR TITLE
Update entire Discovery Server documentation [9979]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: docs/user_manual/getting_started/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
+.. _getting_started:
+
+Getting started
+###############
+
+This section explains the basic concepts of the Discovery Server discovery mechanism.
+For more information on the Discovery Server mechanism, please refer to the
```
Comment (richiware) [inline docs/user_manual/getting_started/getting_started.rst:7]: Broken link

File: docs/user_manual/usage/command_line.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    .. code-block:: bash
+
+        $ source <path/to/discovery-server-ws>/discovery-server-ws/install/setup.bash
+        $ discovery-server-X.X.X(d) config_file.xml
+
+-   **Windows**
+
+    .. code-block:: bash
+
+        > . <path/to/discovery-server-ws>/discovery-server-ws/install/setup.bat
```
Comment (richiware) [inline docs/user_manual/usage/command_line.rst:26]: Use Windows slashes

File: docs/xml_profiles/transports.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
+.. _xml_configuration_examples:
+
+Transport protocol configuration
+################################
+
+UDP settings
+************
+
+The XML basically mimics the `UDP attribute C++ source code <cpp_udp_settings>`_:
```
Comment (richiware) [inline docs/xml_profiles/transports.rst:9]: Broken link locally.

File: docs/xml_profiles/transports.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+.. note::
+
+    ``leaseDuration`` is set to ``INFINITY`` in order to mimic the HelloWorldExample participants but can be
+    whatever value without affecting the discovery operation.
+
+TCP settings
+************
+
+The XML basically mimics the `TCP attribute C++ source code <cpp_tcp_settings>`_:
```
Comment (richiware) [inline docs/xml_profiles/transports.rst:27]: Broken link locally.

**Review by richiware — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "restructuredtext.confPath": "${workspaceFolder}/docs"
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply .vscode/settings.json:2]: No it shouldn't be there

File: docs/installation/installation_windows.rst [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    Choose the build configuration by declaring ``CMAKE_BUILD_TYPE`` as Debug or Release.
+    For this example, the Debug option has been chosen, which would be the choice of advanced users for debugging
+    purposes.
+    If using a multi-configuration generator like Visual Studio we recommend to build both in debug and release modes
+
+    .. code-block:: bash
+
+        > colcon build --base-paths src \
+                --packages-up-to discovery-server \
+                --cmake-args -DTHIRDPARTY=ON -DLOG_LEVEL_INFO=ON \
```
Comment (richiware) [inline docs/installation/installation_windows.rst:200]:
```suggestion
                --cmake-args -DLOG_LEVEL_INFO=ON -DCOMPILE_EXAMPLES=ON \
```

File: docs/installation/installation_windows.rst [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+                        -DCOMPILE_EXAMPLES=ON -DINTERNALDEBUG=ON -DCMAKE_BUILD_TYPE=Debug
```
Comment (richiware) [inline docs/installation/installation_windows.rst:201]:
```suggestion
                        -DINTERNALDEBUG=ON -DCMAKE_BUILD_TYPE=Debug
```

File: docs/installation/installation_windows.rst [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+        > colcon build --base-paths src \
+                --packages-up-to discovery-server \
+                --cmake-args -DTHIRDPARTY=ON -DCOMPILE_EXAMPLES=ON -DCMAKE_BUILD_TYPE=Release
```
Comment (richiware) [inline docs/installation/installation_windows.rst:204]:
```suggestion
                --cmake-args -DCOMPILE_EXAMPLES=ON -DCMAKE_BUILD_TYPE=Release
```

File: docs/user_manual/usage/command_line.rst [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+-   **Linux**
+
+    .. code-block:: bash
+
+        $ source <path/to/discovery-server-ws>/discovery-server-ws/install/setup.bash
+        $ discovery-server-X.X.X(d) config_file.xml
+
+-   **Windows**
+
+    .. code-block:: bash
```
Comment (richiware) [inline docs/user_manual/usage/command_line.rst:24]:
```suggestion
    .. code-block:: batch
```

File: docs/xml_profiles/transports.rst (hunk 3 above)
Comment (richiware) [reply docs/xml_profiles/transports.rst:9]:
```suggestion
The XML basically mimics the :ref:`UDP attribute C++ source code <cpp_udp_settings>`:
```

**Review by richiware — CHANGES_REQUESTED** [review]

File: docs/installation/installation_windows.rst (hunk 6 above)
Comment (rsanchez15) [reply docs/installation/installation_windows.rst:200]: Compile examples would be duplicated

File: docs/installation/installation_windows.rst (hunk 7 above)
Comment (rsanchez15) [reply docs/installation/installation_windows.rst:201]: Okey this resolves the above duplicity

**Review by richiware — APPROVED** [review]

**Pull request merged**